File: neetcode150/medium/cpp/53-Maximum-Subarray.cpp
#include <vector>

using namespace std;

// Solution 1
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int tot = 0;
        int best = nums[0];
        for(int n : nums){
            if(n > tot && tot < 0){
                tot = 0; 
            }
            tot += n;
            best = max(tot, best);
        }
        return best; 
    }
};

File: neetcode150/medium/53-Maximum-Subarray.py
# Solution 1:
class Solution:
    def maxSubArray(self, nums):
        tot = 0 
        best = nums[0] 
        for i in nums:
            if i > tot and tot < 0:
                tot = 0
            tot += i
            best = max(best, tot)
        return best
        